lib/trlbuf.c

```diff
@@ -566,51 +501,39 @@ void ResetTrlPreface (void) {
     trlbuf.preface = NULL;
 
     /* Now, allocate new pointer as before */
-    trlbuf.preface = realloc (NULL, INIT_LEN);
+    trlbuf.preface = realloc (NULL, initLength);
 
     /* ...and initialize it to NULL */
     trlbuf.preface[0] = '\0';
 }
-
-/* ------------------------------------------------------------------*/
-/*                          ResetTrlBuf                              */
-/* ------------------------------------------------------------------*/
-
-
-/*
-    Clear out messages from buffer.  Intended for use after
-    messages were already written to file.  Assumes trlbuf.buffer
-    already exists!
-*/
-static void ResetTrlBuf (void) {
+static void ResetTrlBuf (void)
+{
+    /*
+        Clear out messages from buffer.  Intended for use after
+        messages were already written to file.  Assumes trlbuf.buffer
+        already exists!
+    */
 
     extern int status;
-    void asnmessage (char *);
 
     free(trlbuf.buffer);
```

Comment:
@mdlpstsci an inline comment looks like this. Hover the cursor over the line, a blue '+' tab should appear, click on it and enter comment. It is not available for greyed-out code.

Reply:
Thanks for spotting this.

Reply:
In fact, this is quite a bug, and the ``free()`` should be removed altogether due to the subsequent ``realloc()``.

Reply:
Actually, this is odd. It frees and then subsequently re-allocates (mallocs) via a ``realloc(NULL, size)``. I'm not sure of what the point of this could be.

Reply:
I think this is better if the ``free(trlbuf.buffer)`` is removed and just reallocated.
```ctrlbuf.buffer = realloc(trlbuf.buffer, initLength);```

Reply:
Ok so this is why.
```c
/* This function will reset the preface to a null string, so nothing
        more gets written out to any other trailer files.  It also
        reallocates the space to preface so that it doesn't take any memory.
    */
```
COmment in ``void ResetTrlPreface (void)``

Reply:
Or at least a description of the intention. It still seems wrong to me.